java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -1497,6 +1497,31 @@ public final ColumnVector toTitle() {
     assert type.equals(DType.STRING);
     return new ColumnVector(title(getNativeView()));
   }
+
+  /**
+   * Returns a column of capitalized strings.
+   *
+   * If the `delimiters` is an empty string, then only the first character of each
+   * row is capitalized. Otherwise, a non-delimiter character is capitalized after
+   * any delimiter character is found.
+   *
+   * Example:
+   *     input = ["tesT1", "a Test", "Another Test", "a\tb"];
+   *     delimiters = ""
+   *     output is ["Test1", "A test", "Another test", "A\tb"]
+   *     delimiters = " "
+   *     output is ["Test1", "A Test", "Another Test", "A\tb"]
+   *
+   * Any null string entries return corresponding null output column entries.
+   *
+   * @param delimiters Used if identifying words to capitalize. Should not be null.
+   * @return a column of capitalized strings. Users should close the returned column.
+   */
+  public final ColumnVector capitalize(Scalar delimiters) {
+    assert type.equals(DType.STRING);
```

Comment:
If we're bothering to assert the column view type then it would be consistent to also assert the scalar type as well.  Given the native code is blindly casting the scalar handle to a `string_scalar`, I'd be tempted to upgrade the check from an assert to an always check and throw IllegalArgumentException if it's the wrong type.  Debugging a bad pointer type cast in native code is never fun.

Reply:
nice catch, updated